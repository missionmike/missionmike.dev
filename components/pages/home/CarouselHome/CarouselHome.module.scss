@import 'styles/variables';

// Main component.
.carousel {
  overflow: visible;
  margin-top: 2rem;

  // Inner container, contains all slides.
  :global .carousel-inner {
    overflow: visible;
  }

  // Individual slide items, includes captions.
  :global .carousel-item {
    display: block;
    transition: opacity 200ms ease;
    border-radius: 50px;
    box-shadow: 3px 3px 13px rgba(0, 0, 0, 0.25);
    line-height: 0;
    overflow: visible;

    &.active {
      transform: rotate(-10deg);
      z-index: 9;

      .carousel-caption {
        display: block;
        opacity: 1;
        transform: translateX(-3rem) rotate(10deg);
      }

      img {
        opacity: 1;
      }
    }

    &.active + .carousel-item {
      transform: rotate(-5deg);
    }

    img {
      transition: opacity 100ms ease;
      border-radius: 50px;
      opacity: 0.5;
    }

    .carousel-caption {
      line-height: 1.7rem;
      text-align: justify;
      transition: none;
      opacity: 0;
      transform: rotate(10deg);
      right: 0;
      left: 0;

      h3,
      p {
        color: black;
      }
    }
  }

  :global .carousel-control-prev,
  :global .carousel-control-next {
    z-index: 11;
    background-color: black;
    border: 3px solid $pink;
    border-radius: 100%;
    width: 50px;
    height: 50px;
    top: calc(50% - 25px);
    transition: opacity 200ms ease;
    opacity: 0.75;

    &:hover {
      opacity: 1;
    }
  }

  :global .carousel-control-prev {
    transform: translateX(-2rem);
  }

  :global .carousel-control-next {
    transform: translateX(2rem);
  }

  // Dot navigation indicators.
  :global .carousel-indicators {
    position: absolute;
    transform: translateY(12rem);
    z-index: 10;

    button {
      background-color: $gray;
      border: 2px solid $blue;
      width: 10px;
      height: 10px;
      border-radius: 100%;
      margin: 0 0.5rem;
      transition: all 500ms ease;

      &.active {
        background-color: $pink;
        border-color: $pink;
      }
    }
  }
}
