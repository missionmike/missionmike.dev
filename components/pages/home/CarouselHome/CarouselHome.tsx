import React, { useState } from 'react';

import Carousel from 'react-bootstrap/Carousel';
import Image from 'next/image';
import styles from './CarouselHome.module.scss';

const CarouselHome = () => {
  const [index, setIndex] = useState(0);

  const handleSelect = (selectedIndex, e) => {
    setIndex(selectedIndex);
  };

  return (
    <Carousel
      activeIndex={index}
      onSelect={handleSelect}
      className={styles.carousel}
      interval={null}
    >
      <Carousel.Item>
        <Image
          src="/static/images/family-gaming.jpg"
          alt="Photo of Mission Mike playing videogames with his three children. Sitting at a computer desk with a green screen behind them."
          width={1200}
          height={900}
        />
        <Carousel.Caption>
          <p>
            Playing video games with my kids and recording our gameplay together is my fav!! Learn
            about our channels Mikey &amp; Daddy and Sleepy Slawth Gaming.
          </p>
        </Carousel.Caption>
      </Carousel.Item>
      <Carousel.Item>
        <Image
          src="/static/images/office-dogs.jpg"
          alt="Photo of Mission Mike and his two office dogs, Maximus and Magnus lying on the floor in front of the desk."
          width={1200}
          height={900}
        />
        <Carousel.Caption>
          <p>
            I spend the workday coding with my trusty office doggos, Maximus (left) and Magnus.
            Sometimes their snoring can be overheard during Zoom meetings.
          </p>
        </Carousel.Caption>
      </Carousel.Item>
      <Carousel.Item>
        <Image
          src="/static/images/family-breakfast.jpg"
          alt="Mike eating breakfast with his wife Yohana and their children."
          width={1200}
          height={900}
        />
        <Carousel.Caption>
          <p>
            My family is my ❤️... Everything I do, I do for them. Their wellbeing is my mission.
          </p>
        </Carousel.Caption>
      </Carousel.Item>
    </Carousel>
  );
};

export { CarouselHome };
