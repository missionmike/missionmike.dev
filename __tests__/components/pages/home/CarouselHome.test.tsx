import { CarouselHome } from 'components/pages/home/CarouselHome/CarouselHome';
import React from 'react';
import { render } from '@testing-library/react';

describe('SocialBar component', () => {
  it('renders the component', () => {
    const { container } = render(<CarouselHome />);
    expect(container).toMatchSnapshot();
  });
});
