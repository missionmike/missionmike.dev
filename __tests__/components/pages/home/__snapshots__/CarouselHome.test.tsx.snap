// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialBar component renders the component 1`] = `
<div>
  <div
    class="carousel carousel slide"
  >
    <div
      class="carousel-indicators"
    >
      <button
        aria-current="true"
        aria-label="Slide 1"
        class="active"
        data-bs-target=""
        type="button"
      />
      <button
        aria-current="false"
        aria-label="Slide 2"
        data-bs-target=""
        type="button"
      />
      <button
        aria-current="false"
        aria-label="Slide 3"
        data-bs-target=""
        type="button"
      />
    </div>
    <div
      class="carousel-inner"
    >
      <div
        class="active carousel-item"
      >
        <img
          alt="Photo of Mission Mike playing videogames with his three children. Sitting at a computer desk with a green screen behind them."
          data-nimg="1"
          decoding="async"
          height="900"
          loading="lazy"
          src="/_next/image?url=%2Fstatic%2Fimages%2Ffamily-gaming.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fstatic%2Fimages%2Ffamily-gaming.jpg&w=1200&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Ffamily-gaming.jpg&w=3840&q=75 2x"
          style="color: transparent;"
          width="1200"
        />
        <div
          class="carousel-caption"
        >
          <p>
            Playing video games with my kids and recording our gameplay together is my fav!! Check out our YouTube channels
             
            <a
              href="https://www.youtube.com/@MikeyNDaddy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Mikey & Daddy
            </a>
             
            and
             
            <a
              href="https://www.youtube.com/@sleepyslawths"
              rel="noopener noreferrer"
              target="_blank"
            >
              Sleepy Slawths
            </a>
            .
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Photo of Mission Mike and his two office dogs, Maximus and Magnus lying on the floor in front of the desk."
          data-nimg="1"
          decoding="async"
          height="900"
          loading="lazy"
          src="/_next/image?url=%2Fstatic%2Fimages%2Foffice-dogs.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fstatic%2Fimages%2Foffice-dogs.jpg&w=1200&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Foffice-dogs.jpg&w=3840&q=75 2x"
          style="color: transparent;"
          width="1200"
        />
        <div
          class="carousel-caption"
        >
          <p>
            I spend the workday coding with my trusty office doggos, Maximus (left) and Magnus. Sometimes I fear their snoring can be overheard during Zoom meetings.
          </p>
        </div>
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Mike eating breakfast with his wife Yohana and their children."
          data-nimg="1"
          decoding="async"
          height="900"
          loading="lazy"
          src="/_next/image?url=%2Fstatic%2Fimages%2Ffamily-breakfast.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fstatic%2Fimages%2Ffamily-breakfast.jpg&w=1200&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Ffamily-breakfast.jpg&w=3840&q=75 2x"
          style="color: transparent;"
          width="1200"
        />
        <div
          class="carousel-caption"
        >
          <p>
            My family is my ❤️... Everything I do, I do for them. Their wellbeing is my mission.
          </p>
        </div>
      </div>
    </div>
    <a
      class="carousel-control-prev"
      href="#"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="visually-hidden"
      >
        Previous
      </span>
    </a>
    <a
      class="carousel-control-next"
      href="#"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="visually-hidden"
      >
        Next
      </span>
    </a>
  </div>
</div>
`;
